streams/src/test/java/org/apache/kafka/streams/kstream/internals/CogroupedKStreamImplTest.java

```diff
@@ -0,0 +1,369 @@
+            new KeyValueTimestamp<>("k3", 2, 500),
+            new KeyValueTimestamp<>("k2", 13, 500)
+        )));
+    }
+
+    @Test
+    public void testCogroupKeyMixedInTopicsTypesandChageTypes() {
+
+        final Consumed<String, Integer> intergerConsumed = Consumed.with(Serdes.String(), Serdes.Integer());
+        final MockProcessorSupplier<String, Integer> supplier = new MockProcessorSupplier<>();
+        final KStream<String, String> test = builder.stream("one", stringConsumed);
+        final KStream<String, Integer> test2 = builder.stream("two", intergerConsumed);
+
+        final KGroupedStream<String, String> groupedOne = test.groupByKey();
+        final KGroupedStream<String, Integer> groupedTwo = test2.groupByKey();
+
+        final KTable<String, Integer> customers = groupedOne.cogroup(STRSUM_AGGREGATOR)
+            .cogroup(groupedTwo, SUM_AGGREGATOR)
+            .aggregate(SUM_INITIALIZER, Materialized.<String, Integer, KeyValueStore<Bytes, byte[]>>as("store1").withValueSerde(Serdes.Integer()));
+
+        customers.toStream().process(supplier);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+            final TestInputTopic<String, String> testInputTopic = driver.createInputTopic("one", new StringSerializer(), new StringSerializer());
+            final TestInputTopic<String, Integer> testInputTopic2 = driver.createInputTopic("two", new StringSerializer(), new IntegerSerializer());
+            testInputTopic.pipeInput("k1", "1", 0L);
+            testInputTopic.pipeInput("k2", "1", 1L);
+            testInputTopic.pipeInput("k1", "1", 10L);
+            testInputTopic.pipeInput("k2", "1", 100L);
+
+            testInputTopic2.pipeInput("k2", 2, 100L);
+            testInputTopic2.pipeInput("k2", 2, 200L);
+            testInputTopic2.pipeInput("k1", 2, 1L);
+            testInputTopic2.pipeInput("k2", 2, 500L);
+            testInputTopic2.pipeInput("k1", 2, 500L);
+            testInputTopic2.pipeInput("k2", 3, 500L);
+            testInputTopic2.pipeInput("k3", 2, 500L);
+            testInputTopic2.pipeInput("k2", 2, 100L);
+        }
+
+        assertThat(supplier.theCapturedProcessor().processed, equalTo(Arrays.asList(
+                new KeyValueTimestamp<>("k1", 1, 0),
+                new KeyValueTimestamp<>("k2", 1, 1),
+                new KeyValueTimestamp<>("k1", 2, 10),
+                new KeyValueTimestamp<>("k2", 2, 100),
+                new KeyValueTimestamp<>("k2", 4, 100),
+                new KeyValueTimestamp<>("k2", 6, 200),
+                new KeyValueTimestamp<>("k1", 4, 10),
+                new KeyValueTimestamp<>("k2", 8, 500),
+                new KeyValueTimestamp<>("k1", 6, 500),
+                new KeyValueTimestamp<>("k2", 11, 500),
+                new KeyValueTimestamp<>("k3", 2, 500),
+                new KeyValueTimestamp<>("k2", 13, 500)
+        )));
+    }
+
+    @Test
+    public void testCogroupKeyMixedAggregators() {
```

Comment:
Not sure what this test covers?

Reply:
it combines different aggregators

Reply:
Well, `shouldCoGroupStreamsWithDifferentInputTypes` above uses two different aggregator already because the input type are different. Can we consider this test redandent?

Reply:
Yes and no. It does use 2 aggregators but they effectively do that same thing. This time since the produce different outputs it is testing that the aggregators were applied to the correct stream. Whereas the other case is get testing if different aggregators work. it might be a trivial distinction with quite a bit of overlap but I think that it serves a purpose.